Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1421,6 +1423,8 @@ extension BrowserViewController: WKNavigationDelegate {
     }
 
     func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
+        UIApplication.sharedApplication().networkActivityIndicatorVisible = false
+
```

Comment:
This is too simple.

For example, when the `WKWebView` fails to load and finishes via `didFailNavigation:`, the network spinner is never stopped.

Also, when a page loads and is put in the background because you open a new tab, the navigation delegate for the `BrowserViewController` is switched around so you will never receive a `didFinish` or `didFail`. So it may be better to do this in the `TabManager` which is a `WKNavigationDelegate` for *all* open tabs.

I think we should use an *atomic counter* to keep track of the state of all open tabs combined. So that multiple tabs can increment and decrement the spinner state *counter* and then `networkActivityIndicatorVisible` can be configured based on that.

Just keep in mind that there are many tabs loading at the same time.